### PR TITLE
Custom net ps modules

### PR DESCRIPTION
**Pull request by remyd1**
### What does this PR do?

Adding custom modules for ps, network and mount.
#### ps

"ps" module includes 3 new functions :
1. psaux => "ps aux" which can be filtered with a regexp
2. lsof => "lsof -c " to get lsof informations on a specific process
3. netstat => "netstat -nap" to get netstat informations corresponding to a specific process
#### network

It includes two new function.
1. iphexval to get the hexa value of an IP address (useful for PXE)
2. ifacestartswith to retrieve the interface(s) label corresponding to an IP range (e.g. "10.0").
#### mount

It just includes one new function to retrieve custom mount points (which are not standard). Can exclude some particular path.
### Tests written?

No

### CODE REVIEW

File: salt/modules/mount.py [hunk 1]
```diff
@@ -960,3 +960,30 @@ def is_mounted(name):
         return True
     else:
         return False
+
+
+def get_othermountpoints(to_ignore=[]):
```
Comment (cachedout) [inline salt/modules/mount.py]: We can't have mutable types as a default kwarg. Here's an explanation of why this is problematic: http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

Comment (cachedout) [inline salt/modules/mount.py]: I understand where you're going with this function but to be honest, I think it's going to be way too hard to maintain this across multiple platforms and systems. There's really no way here to see what this function does without looking at the code itself. If we were to do something like this, my suggestion would be that we use the Filesystem Hierarchy Standard, though even then there are operating systems which don't comply...

File: salt/modules/ps.py [hunk 2]
```diff
@@ -632,3 +632,73 @@ def get_users():
...
+    with the given pattern. It could be just a name or a regular
+    expression (using python search from "re" module).
+
+    CLI Example:
+
+    .. code-block:: bash
+                
+        salt '*' ps.psaux www-data.+apache2
+    '''
+    import re
```
Comment (cachedout) [inline salt/modules/ps.py]: This import should be at the top of the file instead of a late-import in the function.

File: salt/modules/ps.py [hunk 3]
```diff
@@ -632,3 +632,73 @@ def get_users():
...
+    pattern = re.compile(sanitize_name)
+    ps_aux =  __salt__['cmd.run']("ps aux")
+    found_infos = []
+    ret = []
+    nb_lines = 0
+    for info in ps_aux.splitlines():
+        found = pattern.search(info)
+        if found is not None:
+            # remove 'salt' command from results
+            if info.find("salt") == -1:
```
Comment (cachedout) [inline salt/modules/ps.py]: Again, I understand the intention here but this just blindly removes _all_ matches of that string. We need to ensure that we're *only* removing the process matching our PID.

**Comment by cachedout** [comment]
Hi @remyd1 
Thanks for this. I left a few comments in-line. There are also a number of lint error which need to be addressed: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2147/violations/

Thanks.

**Comment by remyd1** [comment]
Hi Mike,

Ok I will take a look at your suggestions. Some fix are easy to do.

Best,
Remy

**Comment by remyd1** [comment]
I added some modifications. Hope it would be ok.

Best,
Remy

**Comment by cachedout** [comment]
Hi @remyd1 Could you please have a look at the lint improvements recommended here: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/2208/violations/

**Comment by remyd1** [comment]
lint errors.

File: salt/modules/mount.py [hunk 4]
```diff
@@ -960,3 +960,38 @@ def is_mounted(name):
         return True
     else:
         return False
+
+
+def find_no_std_mountpoints(list_of_elements=None):
```
Comment (cachedout) [inline salt/modules/mount.py]: Should this be `find_non_std_mountpoints` instead?

**Comment by remyd1** [comment]
Ok, I changed the name of that mount function.

**Comment by cachedout** [comment]
Hi @remyd1 

Most of this is really good. My main concern here is still that `find_non_std_mountpoints` is too subjective. What may be 'non-standard' to you might be standard to somebody else. Different operating systems have different definitions of what might be 'standard', too. Every time we have tried to do something like this, it ends up being something that is endlessly changed to suit individual opinions and in the end nobody is happy.

So, what I would like to propose is that you remove that from this PR and if you need it, you use it in a custom module that you distribute from `/srv/salt/_modules` on your salt master.

Let me know what you thinks. Thanks.

**Comment by cachedout** [comment]
You'll need to be the one to do it unless I completely close this PR. I'll just wait until Monday for you to do it. I appreciate your patience and understanding here. I'm looking forward to having the rest of this. It will be very useful! Thanks.

**Comment by remyd1** [comment]
Hi Mike,

I updated the code.

Best regards,

Remy

**Comment by cachedout** [comment]
Great! Merged. Thanks, @remyd1

**Pull request merged**